### PR TITLE
Rework of UART on MIMXRT1060

### PR DESCRIPTION
**Pull request by morali**
It's cleanup of code and rework of some of the functions. I think that read and write is now more stable and it shouldn't drop any frames, like it with old implementation.

## Motivation and Context

More optimised code, less variables, use of static variables, checks for NULL pointers.

## How Has This Been Tested?

Tested with SerialCommunication project from nanoFramework Samples.

I've tested read and write on LPUART3 port. I could receive and display strings on terminal properly.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Signed-off-by: morali <jeremi.jasinski@gmail.com>

### CODE REVIEW

File: targets/FreeRTOS/NXP/nanoCLR/Windows.Devices.SerialCommunication/win_dev_serial_native_Windows_Devices_SerialCommunication_SerialDevice.cpp [hunk 1]
```diff
@@ -3,668 +3,699 @@
...
+      NANOCLR_SET_AND_LEAVE(CLR_E_BUFFER_TOO_SMALL);
+    }
+
+    /* push data to buffer */
+    size_t bytesWritten = palUart->TxRingBuffer.Push(data, length);
+
+    /* check if all requested bytes were written */
+    if (bytesWritten != length) {
+      /* not sure if this is the best exception to throw here... */
+      NANOCLR_SET_AND_LEAVE(CLR_E_FAIL);
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP/nanoCLR/Windows.Devices.SerialCommunication/win_dev_serial_native_Windows_Devices_SerialCommunication_SerialDevice.cpp]: This one is meant for a general CLR failure!! Too drastic to use here.
Maybe `CLR_E_IO`...

**Comment by morali** [comment]
Thank you for your insight.
I forgot about the coding style, we use tool for auto-format. It would be nice if nf could adapt some tool.
I will fix coding style and replace exception with different one as you suggested. :)

**Comment by josesimoes** [comment]
@morali 

> I forgot about the coding style, we use tool for auto-format. It would be nice if nf could adapt some tool.

 I hear you! That's on the todo list for quite sometime... 😅 

Would like to find a tool that could work both locally and could be launched as a GitHub check or even Azure Pipeline.

Any suggestions?

**Comment by josesimoes** [comment]
@morali 

> I forgot about the coding style, we use tool for auto-format. It would be nice if nf could adapt some tool.

 I hear you! That's on the todo list for quite sometime... 😅 

Would like to find a tool that could work both locally and could be launched as a GitHub check or even Azure Pipeline.

Any suggestions?

**Pull request merged**